grafana/grafana
# Loki: minor backend-dataframe-format changes

this pull request does a couple small improvements to the dataframes generated on the backend. i am doing this in a separate PR, because this causes large diffs (because the test-snapshots change), so i'm trying to make it small so the snapshot-diff is readable.

the changes:

1. field-names got improved. we now use the standard "Time" name for timestamp-fields, "Value" for metric-values, and "Line" for log-values.
2. for metric frame, we set the frame-type metadata in the dataframe
3. when we attach stats to metric requests, previously we attached the stats to every returned dataframe. but this causes a bug: let's say the query-response has 3 dataframes. when you run such a query in `explore`, then open the `Inspector`, you will see the stats 3 times. this PR changes this so that only the first dataframe gets stats.

how to test:
1. make sure you have the feature-flag `lokiBackendMode` enabled
2. run a logs-query in explore, it should appear normally.
3. use the log-row-context in explore, it should work:
   - run a loki query that contains a search for a string... something like `{a="b"} |= "text"`
   - in the returned lines `text` will be highlighted, and when you mouse-over a log-line, you get a "Show context" link at the end of the line
   - click "Show context", a new popup should appear, verify you see multiple log lines in this popup.
4. run a metric query that produces multiple dataframes:
   - if you have a logs-query that returns some data, like `{a="b"}`, and the returned data is in `json`, something like this should work: `count_over_time({a="b"}|json[10s])`
   - verify that you get multiple "lines" in the graph (if this is true, it also means you received multiple dataframes)
   - now click the `[inspector]` button, look at the stats and verify there are roughly 25lines of stats, and they don't repeat
